src/test/java/io/airlift/airline/TestArgConv.java

```diff
@@ -0,0 +1,55 @@
+package io.airlift.airline;
+
+import com.google.common.base.Joiner;
+import static org.testng.Assert.assertThrows;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.expectThrows;
+import org.testng.annotations.Test;
+
+public class TestArgConv {
+    
+    @Test
+    public void testConstHappy() {
+        ArgConverter("const", "12345");
+    }
+    
+    @Test
+    public void testConstBad() {
+        Throwable catched = expectThrows(ParseOptionConversionException.class, () -> {
```

Comment:
Use `org.assertj.core.api.Assertions#assertThatThrownBy` from
```
        <dependency>
            <groupId>org.assertj</groupId>
            <artifactId>assertj-core</artifactId>
            <scope>test</scope>
        </dependency>
```

Reply:
Need to assert the message of the cause. Don't know how to drill down to that with assertj.

Reply:
```
        assertThatThrownBy(() -> {
            ArgConverter("--value-of", "123456");
        }).isInstanceOf(IllegalArgumentException.class)
                .hasMessageContaining("(2) Argument too long.");
```

this is failing, because its asserting the message of the `ParseOptionConversionException`, not its cause.

Reply:
You can do something like:
```java
assertThatThrownBy(() -> ....)
        .isInstanceOf(...)
        .hasMessage("...")
        .satisfies(e -> assertThat(e.getCause())
                .isInstanceOf(...)
                .hasMessage("..."));
```